File: backend/tests/fixtures/security/javascript/vuln_app.js
const express = require("express");
const { exec } = require("child_process");
const app = express();

app.get("/run", (req, res) => {
  const cmd = req.query.cmd;
  exec(cmd, (err, stdout) => {
    res.send(stdout);
  });
});

app.get("/search", (req, res) => {
  const q = req.query.q;
  db.query("SELECT * FROM users WHERE name = '" + q + "'");
  res.send("ok");
});

File: backend/tests/fixtures/security/python/vuln_app.py
import os
from flask import Flask, Markup, request

app = Flask(__name__)
DEBUG = True
API_KEY = "supersecretvalue123"


@app.route("/search")
def search():
    q = request.args.get("q")
    query = "SELECT * FROM users WHERE name = '" + q + "'"
    cursor.execute(query)
    return query


@app.route("/run")
def run():
    cmd = request.args.get("cmd")
    os.system(cmd)
    return "ok"


@app.route("/page")
def page():
    name = request.args.get("name")
    return Markup(name)
